resourcename/matches_test.go

```diff
@@ -75,14 +75,21 @@ func TestMatches(t *testing.T) {
 			test:     "slash prefix in the name",
 			name:     "/shippers/1",
 			pattern:  "shippers/{shipper}",
-			expected: true,
+			expected: false,
 		},
 
 		{
 			test:     "slash prefix in the pattern",
 			name:     "shippers/1",
 			pattern:  "/shippers/{shipper}",
-			expected: true,
+			expected: false,
+		},
+
+		{
+			test:     "root",
+			name:     "/",
+			pattern:  "/",
+			expected: false,
```

Comment:
I had a feeling this would come back to bite us... the root resource name is not part of the spec (we made it up as a way to bind IAM policies globally) - everything in this PR makes sense except for this - if we special-case `resourcename.Matches` to recognize `resourcename.Root` I think we can merge this

at a later point, we might want to re-evaluate "/" and see if we can come up with better way of handling this that is more spec-compliant